packages/client/src/components/form/FormFieldGenerator.tsx

```diff
@@ -947,26 +947,52 @@ class FormSectionComponent extends React.Component<Props> {
     this.props.setFieldTouched(e.target.name)
   }
 
-  setFieldValuesWithDependency = (
+  static getUpdatedValuesAfterDependentFieldEvaluation = (
+    existingValues: IFormSectionData,
+    fields: IFormField[],
     fieldName: string,
-    value: IFormFieldValue
-  ) => {
-    const updatedValues = cloneDeep(this.props.values)
+    value: IFormFieldValue,
+    evalParams: {
+      config: IOfflineData
+      draft: IFormData
+      user: UserDetails | null
+    }
+  ): IFormSectionData => {
+    const updatedValues = cloneDeep(existingValues)
     set(updatedValues, fieldName, value)
     const updateDependentFields = (fieldName: string) => {
-      const dependentFields = getDependentFields(this.props.fields, fieldName)
+      const dependentFields = getDependentFields(fields, fieldName)
       for (const field of dependentFields) {
         updatedValues[field.name] = evalExpressionInFieldDefinition(
           (field.initialValue as DependencyInfo).expression,
           updatedValues,
-          this.props.offlineCountryConfig,
-          this.props.draftData,
-          this.props.userDetails
+          evalParams.config,
+          evalParams.draft,
+          evalParams.user
         )
         updateDependentFields(field.name)
       }
     }
     updateDependentFields(fieldName)
+    return updatedValues
+  }
+
+  setFieldValuesWithDependency = (
+    fieldName: string,
+    value: IFormFieldValue
+  ) => {
+    const updatedValues =
+      FormSectionComponent.getUpdatedValuesAfterDependentFieldEvaluation(
```

Comment:
or a util function? 🤔

Reply:
🤔 

Util function could be a bit more conventional, but for the Events v1 side I think this is fine. It's just important the logic we make can be comfortably translated to Events v2, but in this case the original bug might not exist there.